### ISSUE TITLE
Standard fee too low in Electrum XVG Wallet (?) - Missing features to cancel or resend stuck transaction

### ISSUE THREAD

**Issue by DasSmeagle** [vergecurrency/VERGE#469]
Hi,

I have a bug and/or feature request.  Not 100% sure. But I think, this behavior it is a big confidence-killer in XVG, so it is important.

I sent a transaction from Electrum XVG Wallet 2.4.1 to binance. But I guess Wallet and receiver do not matter here.

My transaction is visible in blockchain explorer, but I got not confirmation after 17 hours.

I added 1 XVG fee. There is no sign to an avarage user, that 1 XVG, the default in the wallet, may be not enough to get the tranaction through?

https://verge-blockchain.info/tx/f060642c1cce1b1296a127e19a49fc22215b9f73aafaca31843123adf33b022b

Not sure, if this is a bug in wallet or Verge - but it kills the trust in Verge currency, should really be addressed. 

Normally the Electrum-XVG wallet adds 1 XVG automatically. Even if I set the fee higher in preferences (I tested it set to 4) it STILL only adds 1 XVG fee. So how should I know 1 XVG is not enough and it will get stuck?

I did a test adding 2 XVG fee manually - wallet told me "unusually high fee". So again a sign, that normally 1 XVG fee should be enough!? - I accepted an sent that test with 2 xvg fee. And that one went through in minutes.

https://verge-blockchain.info/tx/79e48d4a5014ceb734a62d60c6580ff0682ae3a32bd3a91900148363f7ff4f43

So it seems, the 1 XVG which is added automatically is not enough? 

How can I fix the stuck transaction? There needs to be a feature to cancel/resend or up the fee for stuck transactions...

Or if there already is a feature like that, I couldn't find it, googling for ~1 hour ;)

Anyway, I think it is VERY important to make the user experience better for this kind of problem. User sends money and it seems it is "lost in transit" and there is nothing the user can do, no help file, no option, etc..... In my case it was only 240XVG, no really big amount, but still I spent several hours researching for a solution... And found none.

Thanks,
- Oliver

**Comment by Gustafo76** [comment]
Hi Oliver,

I have totally the same issue, same thing, is crazy bug which is killing me as well, however I sent a lot more than you but same thing, I could not have known that fee 1 XVG is not enough, there is no way to know, now I am stucked too without idea how to fix that :( If this cant be fixed XVG cant be trusted...

My transaction ID : 

https://verge-blockchain.info/tx/021adf79a26f7b292f084270e8e12a7ce6e29758d36ede55ba17d1e363253fe0

If anyone would have hint how to solve I would much appreciate

Regards
A.

**Comment by DasSmeagle** [comment]
@greybushcrypto That Transaction is fine, 1657 confirmations at this moment.

**Comment by greybushcrypto** [comment]
it's not fine....it hasn't been received at the other side in 16hrs.

**Comment by Gustafo76** [comment]
Just for my information, if no confirmation is given the balance comes back to its sender origin if I am not mistaken? If so in what time frame anyone knows ? Probably the aim is to exchange Electrum for Verge wallet entirely so basically Electrum wallet will not be for future supported by updates etc? 
A.

**Comment by DasSmeagle** [comment]
@greybushcrypto I only said that transaction is fine. If it is not visble in the target wallet, it's a problem with that wallet. Not with the Verge blockchain. The funds have been transfered. So you may have a problem, but it is a very different problem.

**Comment by Gustafo76** [comment]
UPDATE:
for my case issue with reciaval closed, wallet just been credited, so on that side all ok, but still is needed to address the issue with the fee confusion.

P.S. Oliver I see your transaction is still pending but seems was sent in the same principle as I did, maybe has something to do with value of total transaction, cant say.
A.

**Comment by Gustafo76** [comment]
UPDATE 2 : 
Weird thing, Verge wallet (recieve side) again put the incomming transaction to "unconfirmed" while electrum wallet (original sender) got back to "unconfirmed" status as well, despite confirmations on blockchain (now 80)

probably similar issue as greybushcrypto has?
A.

**Comment by Gustafo76** [comment]
UPDATE 3:
Original source Electrum wallet been credited back for full amount (despite what is in blockchain recorded) and is going up and down in balance, 1 minute showing is unconfirmed transaction out to verge wallet, other minute says that balance is full in original balance (electrum sending source wallet) as was nothing ever sent, I do not think is issue of wallet itself. Really something developers should look into.

A.

**Comment by Gustafo76** [comment]
UPDATE 4:
Electrum wallet showing "pending" sending out transaction, 0 confirmations (in electrum wallet) while Verge wallet still sees it as unconfirmed incomming transaction. Kind of stucked.

https://verge-blockchain.info/tx/021adf79a26f7b292f084270e8e12a7ce6e29758d36ede55ba17d1e363253fe0
anyone has any ideas please ? Any help appreciated.

Seems trigger of this issue was really too low fee (1 XVG)
Thx
A.

**Comment by radeksalomon** [comment]
Any news from the Verge team? This behaviour is very weird and dangerous. We cannot move our coins because it could be blocked somewhere in the middle. 👎

**Comment by kpatsax** [comment]
Are you using Windows Electrum or Windows Tor Electrum?

**Comment by ryanprovostofc** [comment]
I am also running into the same problem as above with the electrum wallet.  Running it on mac, made the transaction from electrum wallet over to Bittrex about 3 days ago and it is still saying pending/unverified.  I used 1 XVG for the transaction fee, I'm thinking it might be too low but not certain this is the issue.

My transaction ID is 8596265675db193051be6d23060c554e7afb8922768d973f38e4827ad7eea497

If anybody finds a solution to this please post,  getting kind of worried.

Thanks

**Comment by DasSmeagle** [comment]
@kpatsax I am not using the Tor version. I use Electrum-XVG 2.4.1. But I don't think the problem is in the wallet... The transaction is "in the system". It is visible in blockchain. Just no confirmation.

Still the same:

https://verge-blockchain.info/tx/f060642c1cce1b1296a127e19a49fc22215b9f73aafaca31843123adf33b022b


Expected behavior:

- Wallet should inform you that 1 XVG may be too low. (Instead it told me, fee is "unusually high" when I manually set 2 XVG in a later test)
- Or, if 1 XVG should really be enough, someone should look into why these examples don't get confirmations with 1 XVG fee
- There needs to be a way to manually "repair" this. Like cancel a transaction or higher fee.
- Or it should automatically repair itself, like deleting/forgetting the transaction after 48hours.
- If a transaction is "stuck". There needs to be some information for the user. about manually or automatic ways to fix this.

**Comment by benammann** [comment]
wait for block 1807000 and see what happens

**Comment by DasSmeagle** [comment]
Thanks @benammann , but do you know something will happen? Or do you just suggest I should have some patience?

**Comment by rs12123** [comment]
I have the same issue. I'm using the Electrum 2.4.1 Windows wallet. Transaction is unconfirmed, and have zero confirmation. I tried to export my private keys to the Verge Core application to re-send it from there but that didn't work either. Any other ideas?

Transaction: https://verge-blockchain.info/tx/7a83acb7c81f8abf86fb91b795640397f353be9c691e5ba0e01f16e3d57149b8/

**Comment by DasSmeagle** [comment]
@benammann we are at Block 1807126 now. Still no change. Transactions is 5 days old now.

https://verge-blockchain.info/tx/f060642c1cce1b1296a127e19a49fc22215b9f73aafaca31843123adf33b022b

:( I think, this is a serious problem. Not for my case (only 230xvg) but in general. Transaction should never be stuck. Even IF fee is low. It should be robust enough to handle cases like this.

- Oliver

**Comment by DasSmeagle** [comment]
@Gustafo76 goo to hear... How long did it take all in all? 

When you go to "vergecurrency.com" the very first and most prominent thing, right on startpage, is a button: "Download Verge Windows Electrum Wallet"

Not only is it "supported" - at least by the maintainers of web site - but it's the **only** Wallet offered on that start page. It absolutely HAS TO WORK.

- Oliver

**Comment by Gustafo76** [comment]
It took 4 days to complete, yes agree “marketing” must be improved, for now verge is more “tech” coin, but I noticed they try to catch up with these issues. I stopped to use electrum fully now on.

**Comment by DasSmeagle** [comment]
OK, I just started day 5. So there is hope ;)

I wonder why there is no reaction from some "tech guy" ;) Isn't this the right place to get support on stuff like this?

**Comment by Gustafo76** [comment]
Have to confirm some sort of issue still exist, my balance for this transaction keeps poping up and down from confirmed status to unconfirmed in core wallet 4.0.1, no matter how many confirmations is visible by blockchain, anyone experiencing the same? 

TID

https://verge-blockchain.info/tx/f8622c4eda36d23ee045b22d251ca2112997e73aeea7a3597640c7e0de4abaee

Tech info would be appreciated, maybe still due to pools? Using only core wallet.

A.

**Comment by ryanprovofc** [comment]
@Gustafo76 Did your issue just resolve itself over time?  Or was it something you did to make the confirmed transaction go through?  You mentioned something about the pools updating, not sure what that means if you could elaborate. Thanks!

**Comment by JoepW01** [comment]
Hi, I'm having the same problem.. sadly i'm actually sending quite a big amount of coins. 
Is there anyone who can help with this problem? I'm getting kind of freaked out to be honest. 

https://verge-blockchain.info/tx/152ec13ea6f576361d51cbf8a722cfee3e82d271716b897509f6c72a56613bec

**Comment by DasSmeagle** [comment]
Hallo, is there a developer reading?

Even if you cannot help with the stuck transaction, can you tell me:

Is it just because of 1 XVG was not enough fee or is there a different problem? I want to send higher amounts soon from my Electrum Wallet and don't want to hav this happen again...

Thanks,
- Oliver

**Comment by Gustafo76** [comment]
My issue also not resolved, besides transaction is confirmed on blockchain my ballance in core (4.0.1)wallet is jumping up and down from confirmed to unconfirmed, weird, nothing I tried worked to resolve it. As I see others face issue of not getting transaction confirmed still for some days. This I had the same (seems due to small commison but given automatically by Electrum wallet), then after 4 days or so was confirmed and balance came to core wallet, however since then is jumping up and down, friend of mine tried to restore wallet by personal keyes etc, it did nto solve that issue. Anyone with more tech skills might help how to resolve ? Maybe wallet issue which can be fixed by console commands ? 


A.

**Comment by benammann** [comment]
please have a look in #490 ive written a small tutorial how to fix the stuck tx

**Comment by DasSmeagle** [comment]
@benammann Thanks. No way to fix this without going to QT Wallet?

@justinvforvendetta Sorry to bother you, but I think this is a problem which should be fixed. "Stuck" Transactions should be dropped from all mempools after some time. Or an option to trigger this by the user (maybe after a delay?).

**Comment by DasSmeagle** [comment]
Hi,

yeah, problem is fixed for me.

"Suddenly" the transaction was no longer visible on verge-blockchain.info. So I guess, after some time it's delteted from memorypool?

I went offline and started my elexctrum wallet.

There I saw the pending transaction. Was hoping I would be able to delete it, but not possible. 

I copied the transaction id of the pending tx and closed wallet.

I edited \AppData\Roaming\Electrum-XVG\wallets\default_wallet: Deleted all "blocks" containing that transaction. Be careful to understand the structure!!!

I restarted Electrum Wallet, online this time. After letting it sync everything is fine! I can respent the coins and no pending transactions.


My guess is: Wait till transaction is dropped. DON'T LET Wallet resend it, remove it from wallet file.

Maybe I could have just deleted it from wallet file before it was removed from mempool? Not sure.


Still I think, this should be an option on the wallet: Like right click / "Try to revoke" -> Removes it from wallet and triggers servers to drop from mempool.
